agenta-backend/agenta_backend/migrations/mongo_to_postgres/migration.py

```diff
@@ -0,0 +1,522 @@
+import json
+import os
+import asyncio
+from datetime import datetime, timezone
+
+from pymongo import MongoClient
+from bson import DBRef
+from sqlalchemy import text
+from sqlalchemy.dialects.postgresql import UUID
+import uuid_utils.compat as uuid
+
+# Assuming agenta_backend.models.db_models contains your SQLAlchemy models
+from agenta_backend.models.db_models import (
+    UserDB,
+    ImageDB,
+    AppDB,
+    DeploymentDB,
+    VariantBaseDB,
+    AppVariantDB,
+    AppEnvironmentDB,
+    AppEnvironmentRevisionDB,
+    TemplateDB,
+    TestSetDB,
+    EvaluatorConfigDB,
+    HumanEvaluationDB,
+    HumanEvaluationVariantDB,
+    HumanEvaluationScenarioDB,
+    EvaluationAggregatedResultDB,
+    EvaluationScenarioResultDB,
+    EvaluationDB,
+    EvaluationEvaluatorConfigDB,
+    EvaluationScenarioDB,
+    IDsMappingDB,
+    AppVariantRevisionsDB,
+)
+
+from agenta_backend.migrations.mongo_to_postgres.utils import (
+    drop_all_tables,
+    create_all_tables,
+    print_migration_report,
+    store_mapping,
+    get_mapped_uuid,
+    generate_uuid,
+    get_datetime,
+    migrate_collection,
+)
+
+from agenta_backend.models.shared_models import TemplateType
+
+tables = [
+    UserDB,
+    ImageDB,
+    AppDB,
+    DeploymentDB,
+    VariantBaseDB,
+    AppVariantDB,
+    AppVariantRevisionsDB,
+    AppEnvironmentDB,
+    AppEnvironmentRevisionDB,
+    TemplateDB,
+    TestSetDB,
+    EvaluatorConfigDB,
+    HumanEvaluationDB,
+    HumanEvaluationScenarioDB,
+    EvaluationDB,
+    EvaluationScenarioDB,
+    IDsMappingDB,
+    EvaluationEvaluatorConfigDB,
+    EvaluationScenarioResultDB,
+]
+
+
+async def transform_user(user):
+    user_uuid = generate_uuid()
+    await store_mapping("users", user["_id"], user_uuid)
+    return {
+        "id": user_uuid,
+        "uid": user["uid"],
+        "username": user["username"],
+        "email": user["email"],
+        "created_at": get_datetime(user.get("created_at")),
+        "updated_at": get_datetime(user.get("updated_at")),
+    }
+
+
+async def transform_image(image):
+    user_uuid = await get_mapped_uuid(
+        image["user"].id if isinstance(image["user"], DBRef) else image["user"]
+    )
+    image_uuid = generate_uuid()
+    await store_mapping("docker_images", image["_id"], image_uuid)
+    return {
```

Comment:
I'd add a lot of assertions here on what image has. I think the main source of errors we would get (especially for cloud) is some inconsistency in the mongodb data. Adding lots of assertions would allow us to catch any bugs very early.